File: frontend/src/types/order.ts
import type { Address, DietType, MealSession, SpiceLevel } from './customer'
export type { Address, DietType, MealSession, SpiceLevel }

export type OrderStatus =
  | 'PLANNED'
  | 'CONFIRMED'
  | 'PREPARING'
  | 'READY'
  | 'ASSIGNED'
  | 'OUT_FOR_DELIVERY'
  | 'DELIVERED'
  | 'FAILED'
  | 'CANCELLED'
  | 'SKIPPED'

export type OrderType = 'SUBSCRIPTION' | 'AD_HOC' | 'BULK'

export interface Order {
  id: string
  order_date: string
  customer_id: string
  customer_name: string
  customer_phone: string
  delivery_address: Address
  meal_session: MealSession
  diet_type: DietType
  quantity: number
  order_type: OrderType
  spice_level: SpiceLevel
  rice_preference?: string
  special_notes?: string
  status: OrderStatus
  assigned_driver_id?: string
  assigned_driver_name?: string
  sequence_number?: number
  delivery_notes?: string
  delivery_charge: number
  total_price: number
  created_at: string
  updated_at?: string
}

export interface SessionStats {
  total: number
  veg: number
  non_veg: number
  ready: number
  delivered: number
}

export interface DailySummary {
  date: string
  total_orders: number
  breakfast: SessionStats
  lunch: SessionStats
  dinner: SessionStats
}

export interface OrdersListResponse {
  items: Order[]
  total: number
  summary: DailySummary
}

export interface Driver {
  id: string
  username: string
  full_name: string
  phone?: string
  email?: string
  role: string
  is_active: boolean
  driving_status: string
  vehicle_type?: string
  vehicle_number?: string
  assigned_deliveries_count: number
  completed_today_count: number
  last_latitude?: number
  last_longitude?: number
  last_location_time?: string
}

export interface OrderCreateInput {
  order_date: string
  customer_id: string
  customer_name: string
  customer_phone: string
  delivery_address: Address
  meal_session: MealSession
  diet_type: DietType
  quantity: number
  order_type?: OrderType
  spice_level?: SpiceLevel
  rice_preference?: string
  special_notes?: string
  delivery_notes?: string
  delivery_charge?: number
  total_price?: number
}

export interface DriverTodayDeliveriesResponse {
  driver_id: string
  driver_name: string
  driving_status: string
  date: string
  total_assigned: number
  pending_count: number
  out_for_delivery_count: number
  completed_count: number
  failed_count: number
  orders: Order[]
}

export interface LiveDriver {
  driver_id: string
  driver_name: string
  phone?: string
  vehicle_type?: string
  vehicle_number?: string
  driving_status: string
  is_connected: boolean
  latitude: number
  longitude: number
  speed?: number
  heading?: number
  last_updated?: string
  active_order?: {
    order_id: string
    customer_name: string
    customer_phone: string
    area?: string
    address_line?: string
    latitude?: number
    longitude?: number
    meal_session?: MealSession
    quantity: number
  }
}

File: backend/app/models/order.py
from datetime import datetime, timezone
from typing import Dict, Any


def order_helper(order: Dict[str, Any]) -> Dict[str, Any]:
    """Formats a MongoDB order document into a standardized dictionary."""
    if not order:
        return {}
    return {
        "id": str(order["_id"]) if "_id" in order else str(order.get("id", "")),
        "order_date": order.get("order_date", ""),
        "customer_id": str(order.get("customer_id", "")),
        "customer_name": order.get("customer_name", ""),
        "customer_phone": order.get("customer_phone", ""),
        "delivery_address": order.get("delivery_address", {}),
        "meal_session": order.get("meal_session", "LUNCH"),
        "diet_type": order.get("diet_type", "VEG"),
        "quantity": order.get("quantity", 1),
        "order_type": order.get("order_type", "SUBSCRIPTION"),
        "spice_level": order.get("spice_level", "NORMAL"),
        "rice_preference": order.get("rice_preference"),
        "special_notes": order.get("special_notes"),
        "status": order.get("status", "PLANNED"),
        "assigned_driver_id": str(order.get("assigned_driver_id")) if order.get("assigned_driver_id") else None,
        "assigned_driver_name": order.get("assigned_driver_name"),
        "sequence_number": order.get("sequence_number"),
        "delivery_notes": order.get("delivery_notes"),
        "delivery_charge": float(order.get("delivery_charge", 0.0)),
        "total_price": float(order.get("total_price", 0.0)),
        "created_at": order.get("created_at", datetime.now(timezone.utc)),
        "updated_at": order.get("updated_at")
    }
